nativeauthenticator/nativeauthenticator.py

```diff
@@ -148,8 +148,11 @@ def get_or_create_user(self, username, pw, **kwargs):
         if username in self.admin_users or self.open_signup:
             infos.update({'is_authorized': True})
 
-        user_info = UserInfo(**infos)
-        self.db.add(user_info)
+        try:
+            user_info = UserInfo(**infos)
+            self.db.add(user_info)
+        except AssertionError:
```

Comment:
What raises an AssertionError?

Reply:
if the email is not validated by the ORM function, it will return an AssertionError. We can see this behavior being tested here: https://github.com/jupyterhub/nativeauthenticator/blob/7b0953c83b4b7d725561808cfa3ab5dbdd97b9a5/nativeauthenticator/tests/test_orm.py#L20